README.md

```diff
@@ -115,6 +115,29 @@ Now, rendering the component like so:
 
 Will pass through two flags: `someCustomProp` and `name`. Note that currently if a static flag and a passed attribute have the same name, the static flag takes priority.
 
+## Transforming flags
+
+Sometimes you might want to pre-process the flags a bit in Javascript before giving them to Elm. For
+example, all the attributes from the DOM are strings, but you might want to make one of them an
+integer:
+
+```js
+elmWebComponents.register('component-with-ports', ComponentWithPorts, {
+  mapFlags: flags => {
+    const someId = parseInt(flags.someId);
+    return Object.assign({}, flags, { someId });
+  },
+});
```

Comment:
do you think this API would be better if mapFlags took an object, where the key was the flag, and the value was the mapper func?

```js
mapFlags: {
  someId: x => parseInt(x),
}
```

I'm not sure if that's better or worse, what do you think?

Reply:
So yes for the case where it's a genuine map. But another usecase is as a dynamic equiv of `staticFlags` (e.g. if you're doing something in Browser.element with the url, you might add `location: window.location` to the flags).

You might also want to rename a key, or do something more crazy. E.g. we have some stuff where it's easier to just JSON-encode something, put it as a string the the attributes, and then parse that out in JS, like:
```html
<some-tag thingy='{"bool":true,"array":[1,2,3]}'></some-tag>
```
```js
elmWebComponents.register('some-tag', Elm.Thing, {
  mapFlags(flags) {
    return JSON.parse(flags.thingy);
  }
});
```

Reply:
Ah that's a good point.

Could have `mapFlags` take either ? And just use `typeof mapFlags === 'function' ? ... : ...`.

Or I think in that case I'd leave it as a function so we're not restricting functionality. I don't feel that strongly about supporting both or just the function version.

Reply:
Supporting it as a function only doesn't prevent adding the object version later on (semver-wise)